walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,105 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+  /**
+   * This algorithm creates a List of all the times at least one attendee is busy, then uses that
+   * List to find all the times that everyone is available The worst-case runtime is O(n^2)
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> attendeesRequested = request.getAttendees();
+    int meetingDuration = (int) request.getDuration();
+
+    // check for no attendees
+    if (attendeesRequested.isEmpty()) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    // check duration of meeting request
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList();
+    }
+
+    List<TimeRange> badTimes = getBadTimes(events, attendeesRequested);
+
+    // list to hold all the available times (everything not in badTimes)
+    List<TimeRange> goodTimes = getGoodTimes(badTimes, meetingDuration);
```

Comment:
nit: maybe call getBadTimes() inside getGoodTimes() to simplify the query() code. I think this could help make the query function easier to understand.

Reply:
I agree, fixed